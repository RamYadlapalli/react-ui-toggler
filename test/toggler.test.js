import React from 'react';
import ReactDOM from 'react-dom';
import Toggler from '../app/toggler';
import { shallow, mount  } from 'enzyme';
import renderer from 'react-test-renderer';

describe('The Toggler app', () => {
  it('Renders toggle', () => {
    const wrapper = shallow(<Toggler active={true}/>);    
    expect(wrapper.find('.toggle-container').length).toEqual(1);
  });
  it('Test tooltip prop', () => {
    const wrapper = mount(<Toggler tooltip='Test'/>);
    expect(wrapper.find('[title="Test"]').length).toEqual(1);
  });
  it('Test onLabel prop', () => {
    const wrapper = mount(<Toggler onLabel='ON'/>);
    expect(wrapper.find('[data-on="ON"]').length).toEqual(1);
  });
  it('Test offLabel prop', () => {
    const wrapper = mount(<Toggler offLabel='OFF'/>);
    expect(wrapper.find('[data-off="OFF"]').length).toEqual(1);
  });
  // it('Test onChange prop', () => {
  //   // const mockFunc = jest.fn();
  //   // const wrapper = mount(<Toggler onChange={mockFunc}/>);
  //   // expect(mockFunc).toBeCalled();

  // const wrapper = shallow(<Toggler/>);
  //  wrapper.instance().onChange = jest.fn();
  //  wrapper.update();
  //  wrapper.find('.toggle-checkbox').simulate('click');
  //  wrapper.instance().onChange();
  //  expect(wrapper.onChange).toHaveBeenCalled();
  // });
  it('Snapshot creation', () => {
    const rendered = renderer.create(
      <Toggler active={true}/>
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
