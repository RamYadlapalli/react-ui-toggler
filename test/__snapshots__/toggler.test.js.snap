// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Toggler app Snapshot creation 1`] = `
<div
  className="toggle-container"
>
  <label
    className="toggler"
    htmlFor="toogler-label"
  >
    <input
      checked={true}
      className="toggle-checkbox"
      onChange={[Function]}
      type="checkbox"
    />
    <div
      data-off=""
      data-on=""
      title=""
    />
  </label>
</div>
`;
